Dockerfile

```diff
@@ -90,31 +94,50 @@ RUN export DEBIAN_FRONTEND=noninteractive && \
  /var/cache/* \
  /var/lib/apt/lists/* \
  /var/log/*
-################################################################################
 
 ################################################################################
-# Install Ruby Gem dependencies.
-# Ruby Gem versions can be overridden per-site by website repo, e.g by:
-# https://rubygems.org/gems/jumbo-jekyll-theme/versions/
-
-# Set global options for `gem` commands
-COPY gemrc /etc/gemrc
 
-# Install Bundler
+# Bundler specification
 ARG BUNDLER_GEM_VERSION=1.17.2
 ENV BUNDLER_GEM_VERSION ${BUNDLER_GEM_VERSION}
 LABEL Bundler=${BUNDLER_GEM_VERSION}
-RUN gem install \
- --no-user-install \
- bundler -v ${BUNDLER_GEM_VERSION}
-
-# Install Jekyll
-# NB: Sass deprecation warning is currently expected. See
-# https://talk.jekyllrb.com/t/do-i-need-to-update-sass/2509
-ARG JEKYLL_GEM_VERSION=3.8.5
-ENV JEKYLL_GEM_VERSION ${JEKYLL_GEM_VERSION}
-LABEL Jekyll=${JEKYLL_GEM_VERSION}
-RUN gem install --no-user-install jekyll -v ${JEKYLL_GEM_VERSION}
+
+# Rake specification (otherwise some of the required gems below won't install)
+ARG RAKE_GEM_VERSION=12.3.2
+ENV RAKE_GEM_VERSION ${RAKE_GEM_VERSION}
+LABEL Rake=${RAKE_GEM_VERSION}
+
+# and install into a single layer
+RUN gem install --no-document \
```

Comment:
Split these to help with layering, i.e. `gem install bundler` immediately after the `LABEL Bundler=` line.

Reply:
Why `--nodocument`? If we end up having dependant images or subsequent `gem install|update` lines, writing `/etc/gemrc` is cleaner and prevents documentation being added.

Reply:
That was precisely why I didn't split them.

Layering them doesn't help. If you decide to change the bundler version but not the rake version, Docker still needs to build from bundler down, so splitting out the layers doesn't really save much.

Reply:
> Why `--nodocument`? If we end up having dependant images or subsequent `gem install|update` lines, writing `/etc/gemrc` is cleaner and prevents documentation being added.

Mostly because adding gemrc isn't necessary (and adds another layer :) ). We're in control of `Dockerfile` so we can easily and readily ensure that `--nodocument` is always used. I think it makes it a lot clearer this way, otherwise you have to actually open the gemrc file to see what is going on.